source/shared/cpp/AdaptiveCardsSharedModel/AdaptiveCardsSharedModelUnitTest/CustomParsingForIOSTest.cpp

```diff
@@ -0,0 +1,116 @@
+#include "stdafx.h"
+#include "CppUnitTest.h"
+#include "TextBlock.h"
+#include <time.h>
+#include <Windows.h>
+#include <StrSafe.h>
+#include "SharedAdaptiveCard.h"
+#include "BaseCardElement.h"
+#include "UnknownElement.h"
+
+using namespace Microsoft::VisualStudio::CppUnitTestFramework;
+using namespace AdaptiveCards;
+using namespace std;
+
+namespace AdaptiveCardsSharedModelUnitTest
+{
+    TEST_CLASS(CustomParsingForIOSTest)
+    {
+    public:
+        TEST_METHOD(CanGetCustomJsonPayload)
+        {
+            std::string testJsonString = 
+            "{\
+                \"$schema\":\"http://adaptivecards.io/schemas/adaptive-card.json\",\
+                \"type\": \"AdaptiveCard\",\
+                \"version\": \"1.0\",\
+                \"body\": [\
+                    {\
+                        \"type\": \"Random\",\
+                        \"payload\": \"You can even draw attention to certain text with color\"\
```

Comment:
>\"payload\": \"You can even draw attention to certain text with color\"\ [](start = 24, length = 72)

The card author shouldn't have to use "payload", they should be able to use whatever properties they want

Reply:
yes, that's correct; i just leaved it as it is; as it works.

---
In reply to: [172369858](https://github.com/Microsoft/AdaptiveCards/pull/1208#discussion_r172369858) [](ancestors = 172369858)